packages/components/pie-textarea/test/component/pie-textarea.spec.ts

```diff
@@ -407,6 +411,180 @@ test.describe('PieTextarea - Component tests', () => {
+                });
+
+                // Act
+                const labelNodes = component.locator('pie-form-label');
+
+                // Assert
+                await expect(labelNodes).toHaveCount(0);
+            });
+
+            test('should not display a maxLength when label is provided but no maxLength', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        label: 'foo label',
+                    } as TextareaProps,
+                });
+
+                // Act
+                const label = component.getByText('foo label');
+                const maxLengthCounterNodes = component.getByTestId('pie-form-label-trailing');
+
+                // Assert
+                await expect(label).toBeVisible();
+                await expect(maxLengthCounterNodes).toHaveCount(0);
+            });
+
+            test('should not display a maxLength when label is provided and maxLength is 0', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        label: 'foo label',
+                        maxLength: 0,
+                    } as TextareaProps,
+                });
+
+                // Act
+                const label = component.getByText('foo label');
+                const maxLengthCounterNodes = component.getByTestId('pie-form-label-trailing');
+
+                // Assert
+                await expect(label).toBeVisible();
+                await expect(maxLengthCounterNodes).toHaveCount(0);
+            });
+
+            test('should display maxLength when label is present and maxLength provided', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        maxLength: 10,
+                        label: 'foo label',
+                    } as TextareaProps,
+                });
+
+                // Act
+                const maxLengthCounter = component.getByTestId('pie-form-label-trailing');
+
+                // Assert
+                await expect(maxLengthCounter).toBeVisible();
+                await expect(maxLengthCounter).toHaveText('0/10');
+            });
+
+            test('should update the displayed maxLength as a user types', async ({ mount }) => {
+                // Arrange
+                const component = await mount(PieTextarea, {
+                    props: {
+                        maxLength: 10,
+                        label: 'foo label',
+                    } as TextareaProps,
+                });
+
+                // Act & Assert
+                const maxLengthCounter = component.getByTestId('pie-form-label-trailing');
+                await expect(maxLengthCounter).toBeVisible();
+                await expect(maxLengthCounter).toHaveText('0/10');
```

Comment:
Will these work for rtl assuming the ordering is different when in RTL mode?

Reply:
The fractions aren't reversed for RTL, as long as `toHaveText` works the same then this should be okay, but it's probably worth testing both!

Reply:
I will add an RTL visual test, but yeah it works fine in RTL from my testing